### PR TITLE
small fix #19530

### LINKED ISSUE #19530: small fix
file: vendor\yiisoft\yii2\widgets\ActiveField.php
   $options['id'] = Html::getInputId($this->model, $this->attribute);

need fix this to
  $options['id'] = $inputID?:Html::getInputId($this->model, $this->attribute);

otherwise, the formation of rules for activeField does not work correctly when specifying the field id

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.
| PHP version      | 8
| Operating system |

### PR DESCRIPTION
**Pull request by kv4nt**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | small fix #19530

### CODE REVIEW

File: tests/framework/widgets/ActiveFieldTest.php [hunk 1]
```diff
@@ -460,7 +460,7 @@ public function testGetClientOptionsWithCustomInputId()
         $actualValue = $this->activeField->getClientOptions();
 
         $this->assertArraySubset([
-            'id' => 'activefieldtestmodel-attributename',
+            'id' => $this->activeField->inputOptions['id'],
```
Comment (arogachev) [inline tests/framework/widgets/ActiveFieldTest.php:463]: It's better to use the result string in tests.

**Review by arogachev — CHANGES_REQUESTED** [review]
:+1:

File: tests/framework/widgets/ActiveFieldTest.php (hunk 1 above)
Comment (kv4nt) [reply tests/framework/widgets/ActiveFieldTest.php:463]: Заменить 'id' => $this->activeField->inputOptions[ 'id' ] на 'id' => 'custom-input-id'  ?

Comment (arogachev) [reply tests/framework/widgets/ActiveFieldTest.php:463]: @kv4nt Yes.

**Comment by arogachev** [comment]
Thanks. Let's wait for tests to pass (just in case). And one more thing - could you update CHANGELOG?

**Comment by kv4nt** [comment]
> Thanks. Let's wait for tests to pass (just in case). And one more thing - could you update CHANGELOG?

Please tell me where and what to add. I will do.

**Comment by arogachev** [comment]
There is a file in repo root called `CHANGELOD.md`. Please add an entry here similar to others.

File: framework/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ Yii Framework 2 Change Log
 2.0.47 under development
 ------------------------
 
+- Bug #19530: Fix specifying the field id for activeField  (kv4nt)
```
Comment (arogachev) [inline framework/CHANGELOG.md:7]:
```suggestion
- Bug #19530: Fix specifying the field id for `yii\widgets\ActiveField` (kv4nt)
```

**Review by arogachev — APPROVED** [review]

**Comment by arogachev** [comment]
Thanks! :+1:

**Pull request merged**